tests/unit/insertTestResult.unitTest.ts

```diff
@@ -701,4 +702,461 @@ describe("insertTestResult", () => {
+        it('should not throw error', () => {
+            let testResult = testResultsPostMock[4]
+            testResult.testStatus = 'cancelled'
+            testResult.odometerReading = null
+            testResult.odometerReadingUnits = null
+            testResult.countryOfRegistration = null
+            testResult.euVehicleCategory = null
+
+            MockTestResultsDAO = jest.fn().mockImplementation(() => {
+                return {
+                    createSingle: () => {
+                        return Promise.resolve(Array.of(testResultsPostMock[4]));
+                    },
+                    getTestNumber: () => {
+                        return Promise.resolve({ testNumber: "W01A00209", id: "W01", certLetter: "A", sequenceNumber: "002" });
+                    },
+                    getTestCodesAndClassificationFromTestTypes: () => {
+                        return Promise.resolve({
+                            linkedTestCode: "wde",
+                            defaultTestCode: "bde",
+                            testTypeClassification: "Annual With Certificate"
+                        });
+                    }
+                };
+            });
+
+            testResultsService = new TestResultsService(new MockTestResultsDAO());
+
+            return testResultsService.insertTestResult(testResult)
+                .then((data: any) => {
+                    expect(data).to.not.be.eql(undefined)
+                })
+                .catch(() => {
+                    expect.fail()
+                })
+        })
+    })
+
+    context('when inserting an HGV test result with fields applicable to this vehicleType', () => {
+        it('should not throw error', () => {
+            let testResult = testResultsPostMock[4]
+
+            MockTestResultsDAO = jest.fn().mockImplementation(() => {
+                return {
+                    createSingle: () => {
+                        return Promise.resolve(Array.of(testResultsPostMock[4]));
+                    },
+                    getTestNumber: () => {
+                        return Promise.resolve({ testNumber: "W01A00209", id: "W01", certLetter: "A", sequenceNumber: "002" });
+                    },
+                    getTestCodesAndClassificationFromTestTypes: () => {
+                        return Promise.resolve({
+                            linkedTestCode: "wde",
+                            defaultTestCode: "bde",
+                            testTypeClassification: "Annual With Certificate"
+                        });
+                    }
+                };
+            });
+
+            testResultsService = new TestResultsService(new MockTestResultsDAO());
+
+            return testResultsService.insertTestResult(testResult)
+                .then((insertedTestResult: any) => {
+                    expect(insertedTestResult).to.not.be.eql(undefined)
```

Comment:
bad test - can match with error text/stack trace. If you're expecting something specific, match against that.